### DISCUSSION TITLE
How to export. ipa files for. app files built using "briefcase build iOS"

### DISCUSSION THREAD

**Discussion by jlij** [beeware/toga#2656] [Q&A]
rm -rf appName
mkdir appName
mkdir appName/Payload
cp -r appName.app appName/Payload/appName.app
cp Icon.png appName/iTunesArtwork
cd appName
zip -r appName.ipa Payload iTunesArtwork

exit 0

The. ipa built using this method displays "structural system mismatch" after installation

**Comment by freakboy3742** [comment]
Briefcase doesn't currently generate IPA files as an output format. It might be possible to add IPA export, but we'd need to investigate the command line option to `xcodebuild` that produces IPA output.

If you want to submit an app to the App Store, you don't need to explicitly create an IPA file; [we've provided instructions in the Breifcase docs](https://briefcase.readthedocs.io/en/stable/how-to/publishing/iOS.html).

**Reply by jlij** [reply]
Is there any other way to achieve high membership fees due to limited ability? Just turn it into. ipa format, and I'll figure out how to solve the rest myself.

**Reply by freakboy3742** [reply]
Sorry - I have no idea what you're asking. What membership fees are you referring to? What availability?

**Reply by jlij** [reply]
I currently do not have the ability to pay for Apple Developer's fees. 
Is there any way to package the project directly into. ipa format?

**Reply by freakboy3742** [reply]
Yes - [it is *possible* get an IPA file out of Xcode](https://stackoverflow.com/questions/5499125/how-to-create-ipa-file-using-xcode). There are also [command line alternatives](https://stackoverflow.com/questions/2664885/xcode-build-and-archive-from-command-line) to these commands. 

However, there will be limits on how you can use the IPA that has been generated - unless the person you're distributing the IPA to is on your development team, or you've got a jailbroken phone, they won't be able to run the app.

**Reply by jlij** [reply]
I have tried your suggestion, but the prerequisite is to have your own signature, which I did not have.
Can you add the export" .ipa" file feature to your project?

**Reply by freakboy3742** [reply]
We can't magically make features that Xcode itself doesn't support. 

While an IPA file is a zip archive, it's not *just* a zip archive. Xcode's export functionality requires a signature, because using apps on iPhone requires signing. The signing is a key part of the process. There's essentially no path to creating a usable app that doesn't involve having at least your own signature.

**Reply by jlij** [reply]
Can you create an unsigned“. ipa” file?
Add signatures through other methods in the future.
There are many tools on the internet for importing signatures in the later stage。

**Reply by jlij** [reply]
rm -rf appName
mkdir appName
mkdir appName/Payload
cp -r appName.app appName/Payload/appName.app
cp Icon.png appName/iTunesArtwork
cd appName
zip -r appName.ipa Payload iTunesArtwork

exit 0

The above command line can convert ". app" to ". ipa" files.
In the later stage, signatures can be successfully added through certain tools, but when installed on the device, a "structural system mismatch" appears。
I feel like it's a command-line issue because it only changes the file encapsulation format。

**Reply by freakboy3742** [reply]
If you believe there is a viable path for making IPAs that aren't signed, go right ahead and use that process. I am not aware of any process of producing an Apple-compatible IPA that doesn't involve using Xcode's official processes.

**Reply by jlij** [reply]
Unsigned IPA files can be installed and used on iOS through the "Troll Store" or "iTools".
Do you know how this type of IPA file is generated?
![Snipaste_2024-09-23_07-55-16](https://github.com/user-attachments/assets/4dddbbf4-7af9-4fbf-b3b0-afd5e35746bc)

**Reply by freakboy3742** [reply]
I have no additional information I can provide here. As I have previously indicated, if you believe that unsigned IPAs can be used, you'll need to work out how to generate them and use them.
